### ISSUE TITLE
EnumConverter doesn't support UInt64 enums properly

### ISSUE THREAD

**Issue by ww898** [dotnet/corefx#31095]
Hi, I got exception on valid input data (.NET Framework 4.7.2 + .NET Core 2.0): 
```
using System;
using System.ComponentModel;

namespace LargeEnumBug
{
    internal static class Program
    {
        private static void Main()
        {
            var c = new EnumConverter(typeof(E));
            var s = (string) c.ConvertTo(null, null, E.V62 | E.V63, typeof(string));
            var e = (E) c.ConvertFrom(null, null, s); // <=== exception is here!!!
        }

        [Flags]
        private enum E : ulong
        {
            V61 = 1ul << 61,
            V62 = 1ul << 62,
            V63 = 1ul << 63
        }
    }
}
```

Exception:
```
Unhandled Exception: System.FormatException: V62, V63 is not a valid value for E. ---> System.OverflowException: Value was either too large or too small for an Int64.
   at System.Convert.ThrowInt64OverflowException()
   at System.UInt64.System.IConvertible.ToInt64(IFormatProvider provider)
   at System.ComponentModel.EnumConverter.ConvertFrom(ITypeDescriptorContext context, CultureInfo culture, Object value)
   --- End of inner exception stack trace ---
   at System.ComponentModel.EnumConverter.ConvertFrom(ITypeDescriptorContext context, CultureInfo culture, Object value)
   at LagreEnumBug.Program.Main() in C:\LargeEnumBug\Program.cs:line 12
```

However, when I change `E.V62 | E.V63` to `E.V62 | E.V61` everything works properly.

**Comment by danmosemsft** [comment]
Seems like a bug. If https://github.com/dotnet/corefx/blob/master/src/System.ComponentModel.TypeConverter/src/System/ComponentModel/EnumConverter.cs#L80 hits an overflow, it should try Uint64. Enum.ToObject supports it.

**Comment by danmosemsft** [comment]
Better, `EnumConverter` can do `Enum.GetUnderlyingType(t)` to determine that it's UInt64.

**Comment by danmosemsft** [comment]
You are welcome to offer a PR, @ww898

**Comment by ww898** [comment]
Some places with the same issue:
https://github.com/dotnet/corefx/blob/master/src/System.ComponentModel.TypeConverter/src/System/ComponentModel/EnumConverter.cs#L99
https://github.com/dotnet/corefx/blob/master/src/System.ComponentModel.TypeConverter/src/System/ComponentModel/EnumConverter.cs#L138
https://github.com/dotnet/corefx/blob/master/src/System.ComponentModel.TypeConverter/src/System/ComponentModel/EnumConverter.cs#L141

**Comment by ww898** [comment]
Made a PR (#31152). Please check it.

**Comment by pjanotti** [comment]
Closed in PR #31152
